### PR TITLE
Simulation/refactor

### PR DESCRIPTION
**Pull request by taweihuang**
#14 Refactor the simulation codes

Results are the same. Just make some codes reusable.

image

### CODE REVIEW

File: simulation/simulation_exp4p.py [hunk 1]
```diff
@@ -1,104 +1,63 @@
...
+
+    # Plot the regret analysis
+    plt.plot(range(times), seq_error, 'r-', label='delta = ' + str(delta_opt))
+    plt.xlabel('time')
+    plt.ylabel('regret')
+    plt.legend()
+    axes = plt.gca()
+    axes.set_ylim([0, 1])
+    plt.title("Regret Bound with respect to T - EXP4P")
+    plt.show()
```
Comment (stegben) [inline simulation/simulation_exp4p.py]: 其實像這種 plot 類型的也可以再包起來做成一個function

**Comment by stegben** [comment]
LGTM

**Comment by ianlini** [comment]
simulation.py應該放進striatum裡面吧

File: simulation/simulation_exp3.py [hunk 2]
```diff
@@ -1,90 +1,41 @@
-import numpy as np
-import matplotlib.pyplot as plt
-import sys
-sys.path.append("..")
 from striatum.storage import history
 from striatum.storage import model
 from striatum.bandit import exp3
+from striatum import simulation as sm
```
Comment (ianlini) [inline simulation/simulation_exp3.py]: This name is bad. You can change to `sim` or just use `simulation`.

**Comment by ianlini** [comment]
The functions in `simulation.py` are not all related to simulation, and it has a very bad structure, but we can refactor it in the future.

File: striatum/simulation.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
+from sklearn.naive_bayes import MultinomialNB
+from sklearn.linear_model import LogisticRegression
+from sklearn.multiclass import OneVsRestClassifier
+import numpy as np
+import matplotlib.pyplot as plt
+
+
```
Comment (ianlini) [inline striatum/simulation.py]: Please add docstrings for all the functions

**Pull request merged**